utxoscanner.go

```diff
@@ -295,6 +295,7 @@ func (s *UtxoScanner) scanFromHeight(initHeight uint32) error {
 scanToEnd:
 	// Scan forward through the blockchain and look for any transactions that
 	// might spend the given UTXOs.
+	var options *rescanOptions
```

Comment:
nice catch! Could we even initiate the options here directly, avoiding the nil check in the loop?

Reply:
I don't think so bc the options can change during the loop if there are any new requests according to `newReqs := s.dequeueAtHeight(height)`

Reply:
Ah. I missed the nilling of the `options` below.

I'm curious though, why does the current version hog up memory? When the previous iteration's `options` go out of scope the GC should be able to free it.

Reply:
The GC should be able to free it when it runs, but I think this PR could reduce heap pressure. WDYT? Memory hogging is just from repeatedly copying the options over and over

Reply:
I see, it is the copying on each iteration that hogs memory faster than the gc can free it probably.

Could an alternative solution be to just pass a reference to the watch list instead of copying it?

Reply:
I don't know if refactoring `rescanOptions.watchList` to `*[][]byte` is worth it for this optimization as there would be a lot of dereferencing in `rescan.go`

Reply:
Remembered that slices are pointers and don't copy the underlying data, the allocations should be from creating the `rescanOptions` struct so many times.